### ISSUE TITLE
Problems caused by #10540

### ISSUE THREAD

**Issue by RinMaru** [RPCS3/rpcs3#10553]
Ratchet and Clank Crack in time and All 4 one seem to crash when Vram gets close to full but theres no error of out of memory in the log

All 4 one is easy to repro just go into the first stage and as Vram builds the crash is inevitable mostly happening at the holo bridge

200% RES Scale and AA is still a problem after #10540 it seems
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/6826709/RPCS3.zip)

#### 5. Please provide your system configuration:
Windows 10
Ryzen 5 5600X
RX 480 4GB
21.5.2
20GB DDR4 RAM

**Comment by Megamouse** [comment]
Try to disable Anti-Aliasing. Also consider these suggestions:
![image](https://user-images.githubusercontent.com/23019877/125870160-5c34d572-f8ee-4e25-8cef-bd9de9a4b8c8.png)

**Comment by RinMaru** [comment]
AA seems to be the issue

**Comment by kd-11** [comment]
Just a note, enable Read Color Buffers in the debug tab, it will fix most of the transparency problems.

**Comment by kd-11** [comment]
This needs quite a bit more work to become stable enough. I know why it fails to work with MSAA and have plans to fix that.

**Comment by kd-11** [comment]
I have updated the title, due to the severity of bugs discovered. Extensive testing is showing a lot of problems are still present. At the very least we should not crash, until the performance has slowed to a crawl.

**Comment by RinMaru** [comment]
> I have updated the title, due to the severity of bugs discovered. Extensive testing is showing a lot of problems are still present. At the very least we should not crash, until the performance has slowed to a crawl.

That crash I reported for crack in time. It did slow down quite a bit before crashing. Once I reached the third fongoid kids with the time loop area the game started chugging

**Comment by kd-11** [comment]
Please retest with https://github.com/RPCS3/rpcs3/pull/10565
I was able to go into pretty extreme overcommit situation with the bugs fixed:
![image](https://user-images.githubusercontent.com/15904127/126041195-ebf3e317-1a2b-41cb-9c2c-629543d54b6e.png)
No random crashing was observed in this situation. Tested by doing the save the child section in ACiT and once I reach the center platform, reload the save and start again. Performance is really, really terrible when memory is strained but crashing seems resolved for the moment.
More work is needed to improve performance under low memory conditions, or alternatively flush data sections to CELL and remove stale objects to improve memory usage. However, this requires 'read buffers' to be always on, which is part of another task which is already in progress (surface cache restructuring)

**Comment by RinMaru** [comment]
> Please retest with #10565
> I was able to go into pretty extreme overcommit situation with the bugs fixed (7.4GB allocated on a 4GB card):
> ![image](https://user-images.githubusercontent.com/15904127/126041195-ebf3e317-1a2b-41cb-9c2c-629543d54b6e.png)
> No random crashing was observed in this situation. Tested by doing the save the child section in ACiT and once I reach the center platform, reload the save and start again. Performance is really, really terrible when memory is strained but crashing seems resolved for the moment.
> More work is needed to improve performance under low memory conditions, or alternatively flush data sections to CELL and remove stale objects to improve memory usage. However, this requires 'read buffers' to be always on, which is part of another task which is already in progress (surface cache restructuring)

yea no longer seeing crashes in that area now. it will slow to a crawl but at least its not crashing and vram seems to clear up as you leave the time loop area. will try All 4 one next

**Comment by schm1dtHL** [comment]
Tested with both the earlier memory improvement build(10540) and the latest master(10565), on both builds Skate 3 either crashes to the desktop or causes a driver timeout when running on my laptop's iGPU( some Vega shit, as part of the Ryzen 5 5600H ). Attached is a log from when it caused a driver timeout on the latest master.
[RPCS3.log](https://github.com/RPCS3/rpcs3/files/6835217/RPCS3.log)

**Comment by kd-11** [comment]
@schm1dtHL You also have a 3050 Ti on the machine. Does that work?
As far as I can tell from the logs, RSX is not even running, so that issue is unrelated to the memory manager.

**Comment by schm1dtHL** [comment]
The 3050 Ti does work just fine. With regards to the iGPU, I did another test on Skate 3, and it did manage to boot and go in-game, so the CTDs do appear to be fixed at least. The performance, however, is lower than what I observed on a build from prior to the memory management PR, with the current build running at around 30fps at the University spawn, whilst the older build ran at around 46-48fps with the same settings(which was higher than what my 3050 Ti obtained on Skate 3 in the same place!). Memory usage on the iGPU seemed to be around half of the 512 dedicated being used, with shared usage varying from 0.5 to 1 gigs.

**Comment by kd-11** [comment]
Thanks. Yes, I think maybe 512M dedicated is just so small and the system keeps panicking thinking we're running out of memory since only a few hundred MBs are left. I'll see what I can do for that. In the meantime, please create a ticket for the issue so that I do not forget. Attach a log file as well.
